### ISSUE TITLE
Links to Mastodon posts are getting mangled

### ISSUE THREAD

**Issue by rickosborne** [bookwyrm-social/bookwyrm#2740]
Totally unfounded guess: it looks like because my username `@rick` is the same on both BW and Mastodon, the magic link-ification breaks.

**Describe the bug**

You can see the end result in this BW post: https://lit.rickosborne.org/user/rick/review/3365

Note the busted link: `<a href="https://ricko.social/ricko.social/ href="https://lit.rickosborne.org/user/rick">@rick/110016877123858954`

The text I entered was:

```
I have thoughts on this book: https://ricko.social/@rick/110016877123858954

tl;dr: It was mostly enjoyable, if a bit thin or low-stakes.  I’d read more by the author.

BackHomeAgain #DualPOV #FamilyIssues #LoveAtFirstSight #MinimalConflict #SecondChance #ThirdPersonPOV
```

Somehow, that URL `https://ricko.social/@rick/110016877123858954` is getting mangled into the abomination `<a href="https://ricko.social/ricko.social/ href="https://lit.rickosborne.org/user/rick">@rick/110016877123858954`.

**Expected behavior**

I'd expect it to linkify the URL correctly.

**Instance**

This is my own self-hosted instance, lit.rickosborne.org running BW 0.5.3.

---

**Desktop (please complete the following information):**
    - OS: macOS Ventura 13.0.1
    - Browser Safari-ish (technically, the Fluid app)

**Comment by mouse-reeve** [comment]
Okay I'm trying to write a unit test to replicate this, but in the mean time, totally as an aside, you can do half stars by clicking a star a second time

**Comment by mouse-reeve** [comment]
I tried to replicate this and so far haven't succeeded. I entered the text you provided (with and without the emoji stars) and got a normally formatted status:
<img width="774" alt="Screen Shot 2023-03-13 at 6 52 23 PM" src="https://user-images.githubusercontent.com/1807695/224871837-cb708bb4-5ee6-4916-8651-c742ea127c00.png">

Can you edit the status (the edit button is in the three dot "more options" menu for the status) and see what the text is in the edit box? That might indicate something about where the link got mangled.

**Comment by rickosborne** [comment]
Interesting!

When I go to Edit, I get exactly the text I originally entered:

```
4¼⭐️/5

I have thoughts on this book: https://ricko.social/@rick/110016877123858954

tl;dr: It was mostly enjoyable, if a bit thin or low-stakes.  I’d read more by the author.

#BackHomeAgain #DualPOV #FamilyIssues #LoveAtFirstSight #MinimalConflict #SecondChance #ThirdPersonPOV
```

When I re-save it (unchanged) it goes back to being mangled.

Could it be because `@rick` is my name in both places?  Like, if you changed `@rick` to `@mouse`, would that show the problem?

**Comment by rickosborne** [comment]
And if you were curious, when I pull the record from `bookwyrm_status`, here's the HTML-formatted version:

```html
<p>4¼⭐️/5</p>
<p>I have thoughts on this book: &lt;a href="https://ricko.social/ricko.social/ href="https://lit.rickosborne.org/user/rick"&gt;@rick/110016877123858954</p>
<p>tl;dr: It was mostly enjoyable, if a bit thin or low-stakes.  I’d read more by the author.</p>

BackHomeAgain #DualPOV #FamilyIssues #LoveAtFirstSight #MinimalConflict #SecondChance #ThirdPersonPOV
```

(Hmm.  I hadn't noticed that missing leading # in the first tag.  I wonder if that's relevant?  Like it's looking for the `#` as part of a URL?  Or maybe it's a red herring.)

**Comment by mouse-reeve** [comment]
Yes! Thank you that is 100% why it was happening for you; when I used @mouse in the url I got the same bug.

**Comment by rickosborne** [comment]
And just for completeness, I tried submitting a manually-created `a` tag, but it also got mangled around my name.

Submitted:

```html
4¼⭐️/5

I have thoughts on this book: <a href="https://ricko.social/@rick/110016877123858954">ricko.social/@rick/110016877123858954</a>

tl;dr: It was mostly enjoyable, if a bit thin or low-stakes.  I’d read more by the author.

#BackHomeAgain #DualPOV #FamilyIssues #LoveAtFirstSight #MinimalConflict #SecondChance #ThirdPersonPOV
```

Rendered:

```html
<p>4¼⭐️/5</p>
<p>I have thoughts on this book: <a href="https://ricko.social/\&lt;a href=">@rick\</a>/110016877123858954”&gt;ricko.social/<a href="https://lit.rickosborne.org/user/rick">@rick</a>/110016877123858954</p>
<p>tl;dr: It was mostly enjoyable, if a bit thin or low-stakes.  I’d read more by the author.</p>

BackHomeAgain #DualPOV #FamilyIssues #LoveAtFirstSight #MinimalConflict #SecondChance #ThirdPersonPOV
```
